aiida_quantumespresso/parsers/cp.py

```diff
@@ -121,15 +122,16 @@ def parse(self, **kwargs):
                     traj_data = parse_cp_traj_stanzas(
                         num_elements=elements, splitlines=data, prepend_name=f'{name}_traj', rescale=scale
                     )
-                    # here initialize the dictionary. If the parsing of positions fails, though, I don't have anything
-                    # out of the CP dynamics. Therefore, the calculation status is set to FAILED.
-                    if extension != 'cel':
+                    # here initialize the dictionary.
+                    if extension == 'cel':
+                        # NOTE: the trajectory output has the cell matrix transposed!!
+                        raw_trajectory['cells'] = numpy.array(traj_data['cells_traj_data']).transpose((0, 2, 1))
+                    elif extension != 'str':
```

Comment:
Think it would be more legible if you also make this a positive condition `elif extension == 'str':` and then have the generic case in the `else`.